src/domain/entities/BasicDataSet.ts

```diff
@@ -0,0 +1,48 @@
+import { Struct } from "$/domain/entities/generic/Struct";
+import { Id, NamedRef } from "$/domain/entities/Ref";
+
+export interface BasicDataSetAttrs {
+    id: Id;
+    translations: D2Translation[];
+    displayName: string;
+    formType: DataSetFormType;
+    attributeValues: AttributeValue[];
+}
+
+export class BasicDataSet extends Struct<BasicDataSetAttrs>() {
+    constructor(attrs: BasicDataSetAttrs) {
+        super(attrs);
+        validateDataSetIsAllowed(this);
```

Comment:
Not wrong, just as a note. It depends on how the model is going to be used. If we want more control over the validation, we set the constructor private and create a `BasicDataSet.create` that returns an Either.